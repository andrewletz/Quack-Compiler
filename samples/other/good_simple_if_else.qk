if true {
    "This should print1\n".PRINT();
} else {
    "This should NOT print\n".PRINT();
}

if false {
    "This should NOT print\n".PRINT();
} elif false {
    "This should also NOT print\n".PRINT();
} else {
    "This should print2\n".PRINT();
}

x = "Hello World";

if x == "Hello World" {
    "This should print3\n".PRINT();
} else {
    "This should NOT print\n".PRINT();
}

if not (x == "Hello World") {
    "This should NOT print\n".PRINT();
} else {
    "This should print4\n".PRINT();
}
