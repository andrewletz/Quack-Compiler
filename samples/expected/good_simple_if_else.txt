This should print1
This should print2
This should print3
This should print4
